src/OidcClient.ts

```diff
@@ -123,42 +123,41 @@ export class OidcClient {
     }
 
     public async readSigninResponseState(url: string, removeState = false): Promise<{ state: SigninState; response: SigninResponse }> {
-        this._logger.debug("readSigninResponseState");
+        const logger = this._logger.create("readSigninResponseState");
 
         const response = new SigninResponse(UrlUtils.readParams(url, this.settings.response_mode));
         if (!response.state) {
-            this._logger.error("readSigninResponseState: No state in response");
-            throw new Error("No state in response");
+            // https://github.com/microsoft/TypeScript/issues/46972
```

Comment:
this issue does not explain for me that code flow

Reply:
yeah it was happening in too many places to include the comment each time, though I may refactor this to a `throw null` on the following line since this workaround feels too hacky